### ISSUE TITLE
Invalid "Subsequent variable declarations must have the same type" error

### ISSUE THREAD

**Issue by mhegazy** [microsoft/TypeScript#583]
``` TypeScript
var eventsListener = (e: any): void => {
    login();
}

// Unexpected Error: Subsequent variable declarations must have the same type.  Variable 'eventsListener' must be of type 'any', but here has type '(e: any) => void'.
var eventsListener = (e: any): void => {
};

var login = () => {
    var x = {
        events_listener: { events: '*', listener: eventsListener }
    };
}
```
